torch/_functorch/partitioners.py

```diff
@@ -863,6 +863,14 @@ def is_fusible(a, b):
             return True
         if can_fuse_into_triton_kernel_wrapper_functional(a, b):
             return True
+        if (
```

Comment:
Totally agreed - I sent this to Richard but I was going to check with you - what do you think of something like this instead?
```
a = recursively_remove_getitems(a)
b = recursively_remove_getitems(b)
return op_types.is_fusible(a) and op_types.is_fusible(b)
```

since as you pointed out, we treat `operator.getitem` as "always fusible", which seems bad (aka any other ops that return tuples of tensors but are not themselves fusible might suffer in a similar way).

In terms of landing order, I was thinking of landing this change first since it's needed to unblock internal and is a bit less risky.

Reply:
I'm fine with landing this first. Out of curiosity, does https://github.com/pytorch/pytorch/pull/126446 also solve this?

Reply:
> Out of curiosity, does https://github.com/pytorch/pytorch/pull/126446 also solve this? Perhaps while also including operator.getitem in the view list?

Hmm it doesn't look like it (my local copy already has that change to always recompute views, and I also tried tweaking the list to include `operator.getitem` in the list of views). My local change:
```
diff --git a/torch/_functorch/partitioners.py b/torch/_functorch/partitioners.py
index 81e2f297f6f..8c02fb68211 100644
--- a/torch/_functorch/partitioners.py
+++ b/torch/_functorch/partitioners.py
@@ -1293,6 +1293,7 @@ def get_default_op_list() -> OpTypes:
         aten.as_strided,
         aten.permute,
         aten.select,
+        operator.getitem,
     ]
     view_ops = recomputable_view_ops
     default_recomputable_ops += [
```

I tried running the same test locally that I have in my PR, and I'm still seeing `sigmoid()` get saved as an activation (even though it could be fused into an existing inductor backward kernel.

Reply:
(my local test):
```
import torch
import triton
import triton.language as tl

def test_triton_kernel_not_fusable_with_users():
    @triton.jit
    def _sin_kernel(
        in_ptr0,
        out_ptr,
        out2_ptr,
        n_elements,
        BLOCK_SIZE: "tl.constexpr",
    ):
        pid = tl.program_id(axis=0)
        block_start = pid * BLOCK_SIZE
        offsets = block_start + tl.arange(0, BLOCK_SIZE)
        mask = offsets < n_elements
        x = tl.load(in_ptr0 + offsets, mask=mask)
        output = tl.sin(x)
        tl.store(out_ptr + offsets, output, mask=mask)
        tl.store(out2_ptr + offsets, output, mask=mask)

    from typing import List

    from torch._library import capture_triton, triton_op

    @triton_op("mylib::sin_kernel", mutates_args={})
    def sin_kernel(x: torch.Tensor) -> List[torch.Tensor]:
        n_elements = x.numel()
        out = torch.empty_like(x)
        out2 = torch.empty_like(x)
        capture_triton(_sin_kernel)[(n_elements,)](
            x, out, out2, n_elements, BLOCK_SIZE=4
        )
        return [out, out2]

    class MySin(torch.autograd.Function):
        @staticmethod
        def forward(ctx, x):
            out, saved = tuple(torch.ops.mylib.sin_kernel(x))
            ctx.save_for_backward(x, saved)
            return out

        @staticmethod
        def backward(ctx, grad):
            (x, saved) = ctx.saved_tensors
            return grad * saved.sigmoid() * x

    @torch.compile(backend="aot_eager")
    def f(x):
        return MySin.apply(x)

    x = torch.randn(4, 4, requires_grad=True, device='cuda')
    out = f(x)

test_triton_kernel_not_fusable_with_users()
```